### PR TITLE
Extend test_RN50_data_pipeline.py test

### PR DESCRIPTION
**Pull request by JanuszL**
- adds an ability to define number of shards and GPU id for a single instance of test

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It adds an ability to define number of shards and GPU id for a single instance of test

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     adds an ability to define number of shards and GPU id for a single instance of test
 - Affected modules and functionalities:
     test_RN50_data_pipeline.py
 - Key points relevant for the review:
     logic behind the new options
 - Validation and testing:
     CI, local testing
 - Documentation (including examples):
     NA

**JIRA TASK**: *[NA]*

### CODE REVIEW

File: dali/test/python/test_RN50_data_pipeline.py [hunk 1]
```diff
@@ -372,4 +387,4 @@ def update(self, val, n=1):
                   .format(pipe_name.__name__, j + 1, iters, data_time.avg, data_time.max_val, N * BATCH_SIZE / data_time.avg))
               end = time.time()
 
-        print("OK {0}/{1}: {2}".format(i + 1, data_set_len, pipe_name.__name__))
+        print("OK {0}/{1}: {2}".format(i + 1, data_set_len, pipe_name.__name__))
```
Comment (jantonguirao) [inline dali/test/python/test_RN50_data_pipeline.py:390]: missing empty line

**Review by jantonguirao — APPROVED** [review]

**Comment by JanuszL** [comment]
!build

File: dali/test/python/test_RN50_data_pipeline.py (hunk 1 above)
Comment (JanuszL) [reply dali/test/python/test_RN50_data_pipeline.py]: Done

File: dali/test/python/test_RN50_data_pipeline.py [hunk 2]
```diff
@@ -226,8 +226,15 @@ def define_graph(self):
                     help='Disable mmap for DALI readers. Used for network filesystem tests.')
 parser.add_argument('-s', '--small', action='store_true',
                     help='use small dataset, DALI_EXTRA_PATH needs to be set')
+parser.add_argument('--number_of_shards', default=None, type=int, metavar='N',
+                    help='Number of shards in the dataset')
+parser.add_argument('--assign_gpu', default=None, type=int, metavar='N',
+                    help='Assign a given GPU, cannot be used with --gpus')
```
Comment (banasraf) [inline dali/test/python/test_RN50_data_pipeline.py:232]:
```suggestion
                    help='Assign a given GPU. Cannot be used with --gpus')
```

File: dali/test/python/test_RN50_data_pipeline.py [hunk 3]
```diff
@@ -226,8 +226,15 @@ def define_graph(self):
...
+parser.add_argument('--assign_shard', default=0, type=int, metavar='N',
+                    help='Assign a given shard id, if used with --gpus it assigns the first GPU to this id and next GPUs get consecutive ids')
```
Comment (banasraf) [inline dali/test/python/test_RN50_data_pipeline.py:234]:
```suggestion
                    help='Assign a given shard id. If used with --gpus, it assigns the first GPU to this id and next GPUs get consecutive ids')
```
?

**Review by banasraf — APPROVED** [review]

**Comment by JanuszL** [comment]
!build

File: dali/test/python/test_RN50_data_pipeline.py (hunk 2 above)
Comment (JanuszL) [reply dali/test/python/test_RN50_data_pipeline.py]: Done

File: dali/test/python/test_RN50_data_pipeline.py (hunk 3 above)
Comment (JanuszL) [reply dali/test/python/test_RN50_data_pipeline.py]: Done

**Pull request merged**